source/grid/manifold_lib.cc

```diff
@@ -2198,7 +2199,7 @@ TransfiniteInterpolationManifold<dim, spacedim>::pull_back(
       // prevent division by zero. This number should be scale-invariant
       // because Jinv_deltaf carries no units and x is in reference
       // coordinates.
-      if (std::abs(delta_x * Jinv_deltaf) > 1e-12)
+      if (std::abs(delta_x * Jinv_deltaf) > 1e-12 && !must_recompute_jacobian)
```

Comment:
What is this change about?

Reply:
It's to avoid an unnecessary computation. While super cheap in comparison, I was confused by this update step when debugging for the case of an invalid Jacobian. The code suggest we update `inv_grad` even though it is completely overridden in the next iteration here:
https://github.com/dealii/dealii/blob/2d712df2abcc2da14253ed41288146364607311a/source/grid/manifold_lib.cc#L2122-L2137